input/docs/handbook/collections/index.md

```diff
@@ -82,14 +82,124 @@ var myBindingOperation = mySource
+{
+    private readonly ObservableAsPropertyHelper<IEnumerable<bool>> _items;
+    public IEnumerable<bool> Items => _items.Value;
+
+    public ViewModel()
+    {
+        var service = new SimpleService();
+        _items = service.Items
+            // Transform the incoming new collections as you wish
+            // using the power of reactive extensions.
+            .Select(items => items.Skip(1))
+            // And finally convert the collection to a property,
+            // declared as ObservableAsPropertyHelper. Remember, 
+            // that for each change a *new* collection is published.
+            // Don't try to mutate it, this won't work. Read further
+            // to learn how to work with mutable collections.
+            .ToProperty(this, x => x.Items);
+    }
+}
+```
+
+### Complex Scenario
+
+```cs
+public class ComplexService 
+{
+    public SourceList<bool> Items { get; } = new SourceList<bool>();
+
+    public Service()
+    {
+        // With DynamicData you can easily manage mutable datasets,
+        // even if they are extremely large. In this complex scenario 
+        // a service mutates the collection, by using .Add(), .Remove(), 
+        // .Clear(), .Insert(), etc. DynamicData takes care of
+        // allowing you to observe all of those changes.
+        Items.Add(true);
+        Items.RemoveAt(0);
+        Items.Add(false);
+    }
+}
+```
+
+There are two approaches to make the items in `SourceList` stay in sync with the UI.
+
+```cs
+public class ViewModel : ReactiveObject
+{
+    private readonly ObservableAsPropertyHelper<IEnumerable<bool>> _first;
+    public IEnumerable<bool> First => _first.Value;
+
+    private readonly ReadOnlyObservableCollection<bool> _second;
+    public ReadOnlyObservableCollection<bool> Second => _second;
+
+    public ViewModel()
+    {
+        var service = new ComplexService();
+        var changes = service.Items
+            // We use the .Connect() operator to connect to
+            // the data set and turn it into a DynamicData monad.
+            // This gives you the ability to use all DynamicData
+            // LINQ-like query operators.
+            .Connect()
+            // Transform in DynamicData works like Select from
+            // LINQ, it observes changes in one collection, and
+            // projects it's elements to another collection.
+            .Transform(x => !x)
+            // Filter is basically same as .Where() operator
+            // from LINQ. See all operators in DynamicData docs.
+            .Filter(x => x);
+
+        // Now you need to update the UI with those Items.
+        // There are two ways of doing this.
+        _first = changes
+            // Turn the DynamicData monad into a simple
+            // IObservable<T>. So now you can bind it
+            // to ReactiveUI ObservableAsPropertyHelper.
+            // Easy! But again, it refreshes the *whole*
+            // collection at once.
+            .ToCollection()
```

Comment:
`ToCollection` primary purpose is only for aggregation, so we shouldn't  be using this as a example in this case.

Reply:
The reason for this, ToCollection() works pretty differently internally, it re-generates the entire list every time while SourceCache/SourceList Bind() does addition/removals etc.

So `ToCollection()` is only meant for aggregation based on operations where you really need a full collection each time as a observable.